accepted/00078-maintenance-windows-web-calendar.md

```diff
@@ -0,0 +1,185 @@
+
+Provide the user with a calendar view that displays information about the maintenance windows associated with the calendar.
+
+# Motivation
+[motivation]: #motivation
+
+### Why are we doing this?
+Currently we display the contents of the ical file uploaded by the users in the calendar details and creation menus. Since it's difficult to draw information about the maintenance windows defined in that file from the pure ical we want to provide this information displayed as a calendar.
+
+### What use cases does it support?
+
+When choosing a maintenance schedule from a maintenance calendar the user needs to know the details of the maintenance windows contained in the calendar.
+
+Typical workflow when creating new maintenance windows:
+
+1. Head to `Schedule` -> `Maintenance Windows` -> `Calendars` -> `Create`
+2. Follow the steps to create a new calendar
+3. Go to `Schedules` to create a new maintenance schedule
+4. Use the information provided in the created calendars details view to pick a maintenance window
+5. Assign systems to the newly created maintenance schedule
+
+### What is the expected outcome?
+
+Replace the plain ical text in the calendar details and creation menu with a visual representation of the maintenance windows contained in the uploaded ical file.
+Use the created calendar view to also display information about a maintenance window in the maintenance schedule details and creation menu. 
+
+# Detailed design
+[design]: #detailed-design
+
+### UI
+
+Following example screenshot should help to provide a basic idea of how the implementation should look like. (created using react-big-calendar)
+
+Month view:
+![Month view](images/00078-month-view.png)
+
+Day view: 
+![Day view](images/00078-day-view.png)
+
+Events belonging to different Maintenance Window should have a different color
+
+To prevent massive scrolling needed within the calendar view `Back` and `Next` buttons
+should navigate to the previous/next months/days that have at least one maintenance window.
+
+The calendar should have a `read-only` mode to prevent manipulation of displayed events but also support it for potential future use.
+
+### Getting data from the backend
+
+A new function to calculate the maintenance windows contained in a maintenance Calendar needs to be implemented. It looks like `calculateUpcomingMaintenanceWindows` in `IcalUtils` can be reused/adapted in this case.
+
+I see 3 ways of calculating the maintenance windows displayed in the calendar
+1. Stick to the current way and calculate the next x (e.g. 10) maintenance windows
+2. Calculate all the maintenance windows within a given period (e.g. one year) and limit the months displayed in the calendar.
+3. Make a async call to the backend to retrieve all the maintenance windows of a month when clicking the `Back` and `Next` buttons. (basically server side pagination)
+
+From my point of view 1 is out. 3 would be the preferred way, but 2 is faster to implement and probably good enough.
```

Comment:
I agree that 1 is out of the game. I'd still prefer 3 even if it gets a bit harder to achieve, because a fixed period will never satisfy all, and there will always be the corner case where one will have problems: say I have a maintenance window of 30 minutes every day and you fix the given period to 1 year, you'll have to fetch 365 maintenance windows and prepare them to be displayed. I believe a single async fetch of the related data view should be the best approach.

Reply:
I agree with Dario, and this is in part why I brought up the ical format above: ical is nice if people want to import it into their own calendar apps etc, but as an API endpoint format, it doesn't really cover our needs well.

Reply:
Agreed. Let's go with option 3. I updated the RFC accordingly